TSPL.docc/ReferenceManual/Declarations.md

```diff
@@ -3327,15 +3327,25 @@ macro <#name#> = <#macro implementation#>
 
 The *macro implementation* is another macro,
 and indicates the location of the code that performs this macro's expansion.
+The code that performs macro expansion is a separate Swift program,
+that uses the [SwiftSyntax][] module to interact with Swift code.
 Call the `externalMacro(module:type:)` macro from the Swift standard library,
 passing in the name of a type that contains the macro's implementation,
 and the name of the module that contains that type.
 
+[SwiftSyntax]: http://github.com/apple/swift-syntax/
+
 Macros can be overloaded,
 following the same model used by functions.
 A macro declaration appears only at file scope.
 
-For more information, see <doc:Macros>.
+For an overview of macros in Swift, see <doc:Macros>.
+
+<!--
+The placement of the 'where' clause in the grammar below is atypical.
+On other declarations, it goes between the name and definition,
+but here it goes at the end after the #externalMacro(...) bit.
+-->
```

Comment:
FWIW, type aliases work the same way:

```swift
typealias CD<T> = Dictionary<T, T> 
  where T: Hashable
```

so I don't know that it's "atypical" per se; it's how we order `=` definitions and where clauses.

Reply:
Thanks for pointing that out.  It was unexpected enough to me that I originally had a marked to confirm it was correct during tech review, but replaced it with this comment after testing it with the compiler.

It might be worth adding a sentence to the Reference > Generics chapter about where a 'where' clause appears.  I'll take a look at that separately.